### ISSUE TITLE
Crash parsing: throw new

### ISSUE THREAD

**Issue by wickedest** [jshint/jshint#3556]
This code results in a parser crash.  I tried debugging it, but the code is too complex.  **Important** note: there is a space after new which is necessary for the failure.
```js
function code(data) {
  if (data.request === 'foo') {
    throw new 
  }
}
```

Results in this exception:
```
TypeError: Cannot read property 'paren' of undefined
    at Object.nud (/home/jamie/piper/jshint/src/jshint.js:2720:12)
    at expression (/home/jamie/piper/jshint/src/jshint.js:969:34)
    at Object.fud (/home/jamie/piper/jshint/src/jshint.js:5575:18)
    at expression (/home/jamie/piper/jshint/src/jshint.js:966:32)
    at statement (/home/jamie/piper/jshint/src/jshint.js:1960:9)
    at statements (/home/jamie/piper/jshint/src/jshint.js:2014:16)
    at block (/home/jamie/piper/jshint/src/jshint.js:2122:13)
    at Object.fud (/home/jamie/piper/jshint/src/jshint.js:4813:13)
    at expression (/home/jamie/piper/jshint/src/jshint.js:966:32)
    at statement (/home/jamie/piper/jshint/src/jshint.js:1960:9)
    at statements (/home/jamie/piper/jshint/src/jshint.js:2014:16)
    at block (/home/jamie/piper/jshint/src/jshint.js:2122:13)
    at doFunction (/home/jamie/piper/jshint/src/jshint.js:3892:5)
    at Object.fud (/home/jamie/piper/jshint/src/jshint.js:4721:13)
    at expression (/home/jamie/piper/jshint/src/jshint.js:966:32)
    at statement (/home/jamie/piper/jshint/src/jshint.js:1960:9)
```

**Comment by jugglinmike** [comment]
Bug confirmed; thanks for the report! Here's a further-reduced test case:

    function f(){throw new}

(In my own testing, the trailing space doesn't appear to be strictly necessary.)

**Comment by jugglinmike** [comment]
Looks like neither the `throw` nor the function are necessary to demonstrate this bug:

    new;

is sufficient.

**Comment by jugglinmike** [comment]
Here's a fix: gh-3558

**Comment by jugglinmike** [comment]
[We released the fix in JSHint version 2.13.1.](https://jshint.com/blog/2021-08-10/release-2-13-1/)
